org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/quickfix/AbstractDeclarativeIdeQuickfixProvider.java

```diff
@@ -83,4 +84,18 @@ public List<DiagnosticResolution> getResolutions(Options options, Diagnostic dia
 		return getResolutions(options, getFixMethods(diagnostic));
 	}
 
+	/**
+	 * Creates a singleton list with only one {@link TextEdit} that replaces the region of the diagnostic with the given
+	 * text
+	 * 
+	 * @param diagnostic
+	 *            the {@link Diagnostic}
+	 * @param text
+	 *            the text
+	 * @return a singleton list with only one {@link TextEdit}
+	 * @since 2.27
+	 */
+	protected List<TextEdit> createTextEdit(Diagnostic diagnostic, String text) {
```

Comment:
Gut feeling:
Given that the list is effectively not able to grow (cannot be used as a starting point for accumulating more fixes), I'd prefer to have a plain `TextEdit` as return type here.

Why did you choose a `List<TextEdit>` ?

Reply:
It returns a list because the generic interface ITextModification expects a list, and not a TextEdit. The method is meant to ease writing simple QFs. The result is the simplified QF that one can write in org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/quickfix/TestLanguageQuickFixProvider.java in this PR.

I tried offering an overloaded accept method in DiagnosticResolutionAcceptor, as in other cases, but for this case, java could not infer the types of the anonymous lambda on the caller side. That is why I when for a factory method like this one.

Reply:
Understandable.